src/plugins/data/public/query/filter_manager/filter_manager.ts

```diff
@@ -161,6 +164,48 @@ export class FilterManager {
     this.handleStateUpdate(mergedFilters);
   }
 
+  /**
+   * Sets new global filters and leaves app filters untouched,
+   * Removes app filters for which there is a duplicate within new global filters
+   * @param newGlobalFilters
+   */
+  public setGlobalFilters(newGlobalFilters: esFilters.Filter[]) {
```

Comment:
@lizozom, I ended up creating and using these 2 new methods. 

The original setFilters behaviour didn't work for my case because of how it works regarding duplicates between appFilters and globalFilters. 

For example, here I specifically needed setGlobalFilters, to remove any existing duplicate app filters. 

Please let me now if you are ok to leave this 2 as methods, otherwise will have to create as separate helper functions instead.

Reply:
Lets add these for now, for completeness sake.
We'll re-think it when working on (hopefully) getting rid of _g altogether :pray: